Sources/Diagnose/RequestInfo.swift

```diff
@@ -70,11 +70,16 @@ public struct RequestInfo {
       "key.offset: "
       Capture(ZeroOrMore(.digit))
     }
-    guard let offsetMatch = requestTemplate.matches(of: offsetRegex).only else {
-      throw ReductionError("Failed to find key.offset in the request")
+    if let offsetMatch = requestTemplate.matches(of: offsetRegex).only {
+      offset = Int(offsetMatch.1)!
+      requestTemplate.replace(offsetRegex, with: "key.offset: $OFFSET")
+    } else {
+      offset = 0
     }
-    offset = Int(offsetMatch.1)!
-    requestTemplate.replace(offsetRegex, with: "key.offset: $OFFSET")
+
+    // If the request contained source text, remove it. We want to pick it up from the file on disk and most (possibly
+    // all) sourcekitd requests use key.sourcefile if key.sourcetext is missing.
+    requestTemplate.replace(#/ *key.sourcetext: .*\n/#, with: "")
```

Comment:
Not an issue because
1. `key.sourcetext` is truncated in os_log most times anyway (when splitting the log message, we only split it on newlines)
2. We do log the in-memory state of the file if a sourcekitd request crashes and when reducing the crash, we use that logged in-memory state of the file for the contents of the file to reduce.